my-app/src/app/modules/courses-page/services/courses-handler.service.ts

```diff
@@ -3,45 +3,48 @@ import {Injectable} from '@angular/core';
 import {COURSE_DATA} from "@assets/mocks/course-data.mock";
 
 import {ICourse} from "@shared/";
+import {BehaviorSubject} from "rxjs";
 
 export interface ICoursesHandlerService {
-  getList(): ICourse[];
+  getList(): BehaviorSubject<ICourse[]>;
 
   getItemById(id: number): ICourse | undefined;
 
-  removeItem(id: number): ICourse[];
+  removeItem(id: number): void;
 }
 
 @Injectable({
   providedIn: 'root'
 })
 
 export class CoursesHandlerService implements ICoursesHandlerService {
-  private initialCourses: ICourse[] = COURSE_DATA;
-  private visibleCourses: ICourse[] = structuredClone(this.initialCourses);
+  private existingCourses$: BehaviorSubject<ICourse[]> = new BehaviorSubject(structuredClone(COURSE_DATA));
 
-  constructor() {
+  public getList(): BehaviorSubject<ICourse[]> {
+    return this.existingCourses$;
```

Comment:
To protect the subject from changes and prevent access to its `next()` method, usually we expose it as an observable: `return this.existingCourses$.asObservable()`

Reply:
Good point, thank you